prov/gni/include/gnix_wait.h

```diff
@@ -68,6 +68,8 @@ struct gnix_fid_wait {
 	struct slist set;
 };
 
+extern struct dlist_entry gnix_nic_list;
```

Comment:
The global gnix_nic_list is defined within gnix_nic.c not gnix_nic.h, within the nic structure is also a gnix_nic_list but that is just the entry point of that object into the global table.